timesketch/frontend/src/components/Sketch/TimeFilter.vue

```diff
@@ -16,113 +16,219 @@ limitations under the License.
           </p>
+        </div>
+
+        <span style="margin-right:10px;">&rarr;</span>
+
+        <div class="field">
           <p class="control">
-            <button v-if="selectedChip" :disabled="!(startDateTime && endDateTime)" class="button is-success is-outlined" v-on:click="update">Update</button>
-            <button v-else :disabled="!(startDateTime && endDateTime)" class="button is-success is-outlined" v-on:click="submit">+ Add time range</button>
+            <input class="input" ref="endInput" v-model.trim="endDateTime" type="text" :disabled="!isSelected('range')" :placeholder="getPlaceholder('range')"
+            v-on:change="endDateTime && formatDateTime()" v-on:keyup.enter="submit()">
           </p>
         </div>
+      </div>
+    </div>
+
+    <!-- Submit button -->
+    <div class="field is-horizontal">
+      <div class="field is-grouped">
+        <p class="control">
+          <button :disabled="!hasAllInputs()" class="button is-success" v-on:click="submit">{{ selectedChip ? 'Update' : 'Create' }}</button>
+        </p>
+      </div>
     </div>
 
   </div>
 </template>
 
 <script>
 export default {
-  props: ['start', 'end', 'selectedChip'],
+  props: ['selectedChip'],
   data () {
     return {
-      startDateTime: this.start,
-      endDateTime: this.end,
-      chip: this.selectedChip
+      startDateTime: '',
+      endDateTime: '',
+      chip: this.selectedChip,
+      radio: 'interval',
+      offsetStart: '',
+      offsetMinus: 5,
+      offsetPlus: 5,
+      intervals: [
+        {'text': 'Second', 'value' : 's'},
+        {'text': 'Minute', 'value' : 'm'},
+        {'text': 'Hour', 'value': 'h'},
+        {'text': 'Day', 'value': 'd'},
+      ],
+      selectedInterval: 'm',
+    }
+  },
+  created: function () {
+    if (!this.chip) {
+      return
+    }
+
+    // Restoring values from the chip
+    if (this.chip.type === 'datetime_range') {
+      this.radio = 'range'
+      let range = this.chip.value.split(',')
+      this.startDateTime = range[0],
+      this.endDateTime = range[1]
+    }
+    else {
+      this.radio = 'interval'
+      let offset = this.chip.value.split(' ')
+      this.offsetStart = offset[0]
+      this.offsetMinus = offset[1].match(/\d+/)[0]
+      this.offsetPlus = offset[2].match(/\d+/)[0]
+      this.selectedInterval = offset[1].match(/[a-zA-Z]+/)[0]
+      this.formatDateTime()
     }
   },
   methods: {
+    hasAllInputs: function () {
+      if (this.isSelected('interval')) {
+        if (this.offsetStart && this.offsetMinus && this.offsetPlus) {
+          return true
+        }
+      }
+      else if (this.isSelected('range')) {
+        if (this.startDateTime && this.endDateTime){
+          return true
+        }
+      }
+      return false
+    },
+    getPlaceholder: function (radioName) {
+      return (this.radio == radioName) ? '2019-07-07T10:00:01' : ''
```

Comment:
Update the suggested placeholder to todays date instead? :)

Reply:
Why not! 👍